File: haskell/02.hs
penultimate           :: [a] -> a
penultimate []        = error "Can't call on list len < 2"
penultimate (a:[])    = error "Can't call on list len < 2"
penultimate (a:b:[])  = a
penultimate (a:xs)    = penultimate xs

File: scala/src/main/scala/02.scala
object q02 {
  def penultimate[A](l: List[A]):A = l match {
    case List()     => throw new RuntimeException("Can't call on list len > 2")
    case List(a)    => throw new RuntimeException("Can't call on list len > 2")
    case List(a,b)  => a
    case _ => penultimate(l.tail)
  }
}
